pkg/az/az.go

```diff
@@ -1,168 +1,62 @@
-type ZoneUsageRule interface {
-	CanUseZone(zone *ec2.AvailabilityZone) bool
-}
-
-// ZonesToAvoidRule can be used to ensure that certain zone IDs aren't used. This can be used
-// to avoid zones that are known to be overpopulated.
-type ZonesToAvoidRule struct {
-	zoneIDs map[string]struct{}
-}
-
-// CanUseZone checks if the supplied zone is in the list of zones to be avoided.
-func (za *ZonesToAvoidRule) CanUseZone(zone *ec2.AvailabilityZone) bool {
-	_, avoidZone := za.zoneIDs[*zone.ZoneId]
-	return !avoidZone
-}
-
-// NewZonesToAvoidRule returns a new ZonesToAvoidRule with the supplied
-// zones set to avoid
-func NewZonesToAvoidRule(zoneIDs []string) *ZonesToAvoidRule {
-	z := map[string]struct{}{}
-	for _, zoneID := range zoneIDs {
-		z[zoneID] = struct{}{}
+func GetAvailabilityZones(ec2API ec2iface.EC2API, region string) ([]string, error) {
+	zones, err := getZones(ec2API, region)
+	if err != nil {
+		return nil, err
 	}
-	return &ZonesToAvoidRule{zoneIDs: z}
-}
 
-func makeDefaultZoneUsageRules(region string) []ZoneUsageRule {
-	zoneIDs, ok := zoneIDsToAvoid[region]
-	if !ok {
-		return nil
+	numberOfZones := len(zones)
+	if numberOfZones < api.MinRequiredAvailabilityZones {
+		return nil, fmt.Errorf("only %d zones discovered %v, at least %d are required", numberOfZones, zones, api.MinRequiredAvailabilityZones)
 	}
-	return []ZoneUsageRule{NewZonesToAvoidRule(zoneIDs)}
-}
 
-// AvailabilityZoneSelector used to select availability zones to use
-type AvailabilityZoneSelector struct {
-	ec2api   ec2iface.EC2API
-	region   string
-	strategy SelectionStrategy
-	rules    []ZoneUsageRule
-}
-
-// NewSelectorWithDefaults creates a new AvailabilityZoneSelector with the
-// default selection strategy and usage rules
-func NewSelectorWithDefaults(ec2api ec2iface.EC2API, region string) *AvailabilityZoneSelector {
-	return &AvailabilityZoneSelector{
-		ec2api:   ec2api,
-		strategy: NewRecommendedNumberRandomStrategy(),
-		rules:    makeDefaultZoneUsageRules(region),
-		region:   region,
+	if numberOfZones < api.RecommendedAvailabilityZones {
+		return zones, nil
 	}
-}
 
-// NewSelectorWithMinRequired create a new AvailabilityZoneSelector with the
-// minimum required selection strategy and usage rules
-func NewSelectorWithMinRequired(ec2api ec2iface.EC2API, region string) *AvailabilityZoneSelector {
-	return &AvailabilityZoneSelector{
-		ec2api:   ec2api,
-		strategy: NewMinRequiredNumberRandomStrategy(),
-		rules:    makeDefaultZoneUsageRules(region),
-		region:   region,
-	}
+	return randomSelectionOfZones(region, zones), nil
 }
 
-// SelectZones returns a list fo az zones to use for the supplied region
-func (a *AvailabilityZoneSelector) SelectZones() ([]string, error) {
-	availableZones, err := a.getZonesForRegion()
-	if err != nil {
-		return nil, err
+func randomSelectionOfZones(region string, availableZones []string) []string {
+	var zones []string
+	desiredNumberOfAZs := api.RecommendedAvailabilityZones
+	if region == api.RegionUSEast1 {
```

Comment:
I'm sorry, my memory is failing me, but why is there a restriction for us-east-1 specifically? I remember you said something about this, but don't remember the detail. Which means, maybe it's worth having a note here, since if I don't remember it now, I won't remember it in 2 months when I look at this code segment again for whatever reason and I would need to add or remove a region. :)